### PR TITLE
Fix #192: Fail to convert nuklear.

### LINKED ISSUE #192: Fail to convert nuklear
fail to convert https://github.com/vurtun/nuklear/blob/master/nuklear.h

nuklear.h:456:8: warning: a type renamed to 'nk_vec2_' due to the collision with the symbol declared in nuklear.h:3621:23
nuklear.h:457:8: warning: a type renamed to 'nk_vec2i_' due to the collision with the symbol declared in nuklear.h:3622:23
nuklear.h:458:8: warning: a type renamed to 'nk_rect_' due to the collision with the symbol declared in nuklear.h:3627:23
nuklear.h:459:8: warning: a type renamed to 'nk_recti_' due to the collision with the symbol declared in nuklear.h:3628:23
nuklear.h:462:8: warning: a type renamed to 'nk_image_' due to the collision with the symbol declared in nuklear.h:3012:13
dstep: an unknown error occurred: core.exception.RangeError@dstep/translator/Output.d(706): Range violation
----------------
??:? _d_arrayboundsp [0x70c83a]
dstep/translator/Output.d:706 void dstep.translator.Output.Output.writeComment(const(dstep.translator.CommentIndex.CommentIndex.Comment)) [0x6e325b]
dstep/translator/Output.d:728 void dstep.translator.Output.Output.comment(const(dstep.translator.CommentIndex.CommentIndex.Comment)) [0x6e338d]
dstep/translator/Output.d:746 void dstep.translator.Output.Output.flushComments(uint) [0x6e34a7]
dstep/translator/Output.d:769 void dstep.translator.Output.Output.flushLocationBegin(uint, uint, uint, bool) [0x6e3570]
dstep/translator/Output.d:797 void dstep.translator.Output.Output.flushLocation(uint, uint, uint, uint, uint, uint, bool) [0x6e3623]
dstep/translator/Output.d:807 void dstep.translator.Output.Output.flushLocation(uint, uint, uint, bool) [0x6e3673]
dstep/translator/Output.d:812 void dstep.translator.Output.Output.flushLocation(const(clang.SourceLocation.SourceLocation), bool) [0x6e36d5]
dstep/translator/Translator.d:103 dstep.translator.Output.Output dstep.translator.Translator.Translator.translateCursors() [0x6fc709]
dstep/translator/Translator.d:117 immutable(char)[] dstep.translator.Translator.Translator.translateToString() [0x6fc88e]
dstep/translator/Translator.d:75 void dstep.translator.Translator.Translator.translate() [0x6fc5a9]
dstep/driver/Application.d:67 int dstep.driver.Application.Application.run().__foreachbody2(std.typecons.Tuple!(immutable(char)[], immutable(char)[], clang.TranslationUnit.TranslationUnit).Tuple) [0x6a69e9]
/home/elvisxzhou/dlang/dmd-2.079.0/linux/bin64/../../src/phobos/std/parallelism.d-mixin-4007:4053 void std.parallelism.ParallelForeach!(std.range.Zip!(immutable(char)[][], immutable(char)[][], clang.TranslationUnit.TranslationUnit[]).Zip).ParallelForeach.opApply(scope int delegate(std.typecons.Tuple!(immutable(char)[], immutable(char)[], clang.TranslationUnit.TranslationUnit).Tuple)).doIt() [0x6a71b1]
??:? void std.parallelism.run!(void delegate()).run(void delegate()) [0x72228b]
??:? void std.parallelism.Task!(std.parallelism.run, void delegate()).Task.impl(void*) [0x721dc7]
??:? void std.parallelism.AbstractTask.job() [0x75098e]
??:? void std.parallelism.TaskPool.doJob(std.parallelism.AbstractTask*) [0x7209bb]
??:? void std.parallelism.TaskPool.executeWorkLoop() [0x720b19]
??:? void std.parallelism.TaskPool.startWorkLoop() [0x720ac0]
??:? void core.thread.Thread.run() [0x7382af]
??:? thread_entryPoint [0x75a57f]
??:? [0x2a76b9]
core.exception.RangeError@dstep/translator/Output.d(706): Range violation
----------------
??:? _d_arrayboundsp [0x70c83a]
dstep/translator/Output.d:706 void dstep.translator.Output.Output.writeComment(const(dstep.translator.CommentIndex.CommentIndex.Comment)) [0x6e325b]
dstep/translator/Output.d:728 void dstep.translator.Output.Output.comment(const(dstep.translator.CommentIndex.CommentIndex.Comment)) [0x6e338d]
dstep/translator/Output.d:746 void dstep.translator.Output.Output.flushComments(uint) [0x6e34a7]
dstep/translator/Output.d:769 void dstep.translator.Output.Output.flushLocationBegin(uint, uint, uint, bool) [0x6e3570]
dstep/translator/Output.d:797 void dstep.translator.Output.Output.flushLocation(uint, uint, uint, uint, uint, uint, bool) [0x6e3623]
dstep/translator/Output.d:807 void dstep.translator.Output.Output.flushLocation(uint, uint, uint, bool) [0x6e3673]
dstep/translator/Output.d:812 void dstep.translator.Output.Output.flushLocation(const(clang.SourceLocation.SourceLocation), bool) [0x6e36d5]
dstep/translator/Translator.d:103 dstep.translator.Output.Output dstep.translator.Translator.Translator.translateCursors() [0x6fc709]
dstep/translator/Translator.d:117 immutable(char)[] dstep.translator.Translator.Translator.translateToString() [0x6fc88e]
dstep/translator/Translator.d:75 void dstep.translator.Translator.Translator.translate() [0x6fc5a9]
dstep/driver/Application.d:67 int dstep.driver.Application.Application.run().__foreachbody2(std.typecons.Tuple!(immutable(char)[], immutable(char)[], clang.TranslationUnit.TranslationUnit).Tuple) [0x6a69e9]
/home/elvisxzhou/dlang/dmd-2.079.0/linux/bin64/../../src/phobos/std/parallelism.d-mixin-4007:4053 void std.parallelism.ParallelForeach!(std.range.Zip!(immutable(char)[][], immutable(char)[][], clang.TranslationUnit.TranslationUnit[]).Zip).ParallelForeach.opApply(scope int delegate(std.typecons.Tuple!(immutable(char)[], immutable(char)[], clang.TranslationUnit.TranslationUnit).Tuple)).doIt() [0x6a71b1]
??:? void std.parallelism.run!(void delegate()).run(void delegate()) [0x72228b]
??:? void std.parallelism.Task!(std.parallelism.run, void delegate()).Task.impl(void*) [0x721dc7]
??:? void std.parallelism.AbstractTask.job() [0x75098e]
??:? void std.parallelism.TaskPool.doJob(std.parallelism.AbstractTask*) [0x7209bb]
??:? void std.parallelism.TaskPool.executeWorkLoop() [0x720b19]
??:? void std.parallelism.TaskPool.startWorkLoop() [0x720ac0]
??:? void core.thread.Thread.run() [0x7382af]
??:? thread_entryPoint [0x75a57f]
??:? [0x2a76b9]

### PR DESCRIPTION
**Pull request by ciechowoj**

### CODE REVIEW

File: dstep/translator/Output.d [hunk 1]
```diff
@@ -702,8 +704,15 @@ D"[0 .. $ - 1]);
 
             foreach (line; lines)
             {
-                indent();
-                buffer.put(line[indentAmount .. $]);
+                if (line.all!isWhite)
```
Comment (jacob-carlborg) [inline dstep/translator/Output.d]: Do we need to deal with Unicode whitespace? I assume `isWhite` is from the `std.ascii` module and that it doesn't deal with Unicode whitespace.

Comment (jacob-carlborg) [reply dstep/translator/Output.d]: There's a `isWhite` function in `std.uni` as well.

Comment (ciechowoj) [reply dstep/translator/Output.d]: "indentAmount" is counted as number of the spaces, so the code should be correct in the sense it shouldn't crash anymore.

Regarding std.uni it's big pain to use as it works on dchar and string consists of utf8 `char`s.

Comment (ciechowoj) [reply dstep/translator/Output.d]: The unicode library in D isn't the wisest design : (

**Pull request merged**